src/js/plugins/plugin.search.js

```diff
@@ -267,38 +211,16 @@ BookReader.prototype.search = function(term = '', options) {
  */
 BookReader.prototype.BRSearchCallback = function(results, options) {
   this.searchResults = results;
-  this.$('.BRnavpos .search').remove();
-  this.$('.BRmobileSearchResultWrapper').empty(); // Empty mobile results
-
-  // Update Mobile count
-  const mobileResultsText = results.matches.length == 1 ? "1 match" : `${results.matches.length} matches`;
-  this.$('.BRmobileSearchResultWrapper').append(
-    `<div class="BRmobileNumResults">
-    ${mobileResultsText} for "${this.searchTerm}"
-    </div>`
-  );
-
-  let firstResultIndex = null;
-
-  results.matches.forEach((match, i) => {
-    const { text, par: [ partial ] } = match;
-    this.addSearchResult(text, this.leafNumToIndex(partial.page));
-    if (i === 0 && options.goToFirstResult === true) {
-      firstResultIndex = this.leafNumToIndex(partial.page);
-    }
-  });
 
   this.updateSearchHilites();
   this.removeProgressPopup();
-  if (firstResultIndex !== null) {
-    this._searchPluginGoToResult(firstResultIndex);
+  if (options.goToFirstResult) {
+    this._searchPluginGoToResult(results.matches[0].par[0].page);
   }
-}
-
-/** @deprecated */
-BookReader.prototype.BRSearchCallbackErrorDesktop = function(results, options) {
-  const $el = $(this.popup);
-  this._BRSearchCallbackError(results, $el, true);
+  // BookReader's wrapper on jQuery's trigger method is broken. Props are not
+  // passed down, since the method only takes two arguments. The actual props
+  // passed is the BookReader instance.
+  $(document).trigger('BookReader:SearchCallback', { results, options, instance: this });
```

Comment:
The comment above explains why I can't use BookReader's wrapper around `$.fn.trigger`. That's what I described to you in our call last week. The view listens to this event and gets the object as a parameter. The search results are the local variable `results`.

Reply:
speaking in javascript, can we move the callback to a variable above so that it keeps scope?

It would be nice to extend `br.trigger` to accept an optional data object as well.

Reply:
The second argument isn't a callback, it's an object passed through to the event listener. In the docs, it's the `[extraParameters]` parameter. https://api.jquery.com/trigger/ That's why I can't use the BookReader wrapper around `$.fn.trigger`. The wrapper calls `$(document).trigger('BookReader:' + name, this, props)`, so the `extraParameters` that the event listener gets is the BookReader instance. I want to pass in the matches, so I can't use the method wrapped around jQuery's method. Likewise, there'd be no way to extend BookReader's wrapper around it, because it will ultimately pass only the BookReader instance through.

Reply:
perhaps we can make the fork inside of `br.trigger` - i am quite cautious of the 2 different ways we will be firing events. i really like the use of data emission here and that is something we should definitely bring into the fold.

@cdrini - interested in your input here.

Reply:
Since we can assume no one uses `props` (since it does literally nothing :) ), maybe we can change it to this?

```js
/**
 * @param {EVENTS} name
 * @param {*} [props]
 */
BookReader.prototype.trigger = function(name, props = this) {
  $(document).trigger('BookReader:' + name, props);
};
```

Reply:
Fine with me. The inner workings of that method are not long for this world anyway if we're going to be removing jQuery.

Reply:
Note that, if anyone using BR relies on that instance being passed through, those event listeners will now break.